### ISSUE TITLE
[rush] Correct workflow for installing dependencies in one of the project

### ISSUE THREAD

**Issue by MartynasZilinskas** [Microsoft/web-build-tools#820]
Working in mono-repo becomes really painful when installing additional dependencies in the project.

```sh
# cd ../../to/repo/root
$ npx rush unlink
# cd ./to/project/directory
$ npm install dependency
# cd ../../to/repo/root
$ npx rush update
```

Is there a better way to install dependency and not to do this repetitively?

rush already handles projects dependencies, why not adding an additional command that installs dependency in the selected project?

**Comment by pgonzal** [comment]
We had been [relying on VS Code](https://rushjs.io/pages/developer/modifying_package_json/) for this, although that feature has seemed to stop working recently for me.

Can you propose a syntax/design for `rush add` or whatever the command would be?

It's probably not much work to implement.

**Comment by MartynasZilinskas** [comment]
I am thinking something similar to this:

```
rush add <dependency> [--save-dev, -D] [--project <project-name>, -p <project-name>]
```
### Optional arguments
`--save-dev`, `-D`: Install as `devDependencies`. (Default saves only to `dependencies`)
`--project`, `-p`: Installs in selected project from rush.json.

## Examples
CWD: `/` (Repository root)
```sh
$ rush add react -p @simplr/react-forms
```

CWD: `/packages/react-forms` (Resolves that you're in `@simplr/react-forms` project.)
```sh
$ rush add react
```

What do you think?

**Comment by nickpape-msft** [comment]
Seems like a relatively good design to me. This would be a relatively straightforward feature to add.

**Comment by pgonzal** [comment]
How important is `[--project <project-name>, -p <project-name>]`?  Seems like in most cases it should just add to the project of the current working folder where the command is invoked.

I'm assuming that  `rush install` should be automatically invoked as part of this command.  When people are adding multiple packages, it might be useful to have a `--skip-install` option to defer that.

**Comment by pgonzal** [comment]
> And for `--skip-install`, will it update `npm-shrinkwrap` file? Or it only updates `package.json`?

The usage scenario I was envisioning might be:
```
$ cd project1
$ rush add react --skip-install
$ cd ../project2
$ rush add react react-dom --skip-install
$ cd ../project3
$ rush add react # <-- the "rush update" operation happens here
```

**Comment by MartynasZilinskas** [comment]
Sounds good to me. 👍

**Comment by pgonzal** [comment]
This will be implemented in https://github.com/Microsoft/web-build-tools/pull/843

**Comment by pgonzal** [comment]
@MartynasZilinskas this was published today as Rush 5.3.0.  Give it a try!

**Comment by olivierlacan** [comment]
Is there any progress towards @MartynasZilinskas's suggestion of this multi-package flow: 

```
rush add -p typescript tslint tslint-language-service
```

Seems odd (and a poor experience) to set up rush with more than a single package when creating a new project.
